src/android/com/adobe/phonegap/push/FCMService.java

```diff
@@ -329,6 +329,10 @@ private void showNotificationIfPossible(Context context, Bundle extras) {
       Log.d(LOG_TAG, "count =[" + badgeCount + "]");
       PushPlugin.setApplicationIconBadgeNumber(context, badgeCount);
     }
+    if (badgeCount == 0) {
+      NotificationManager mNotificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+      mNotificationManager.cancelAll();
+    }
```

Comment:
@chriscant okay, I haven't had a chance to test yet but what I see looks good. Let me confirm my thinking.

if count >= 0 then set the badge number, which is the current behaviour
if count == 0 then clear all the notifications which is new behaviour
if count is not provided in the push payload then extractBadgeCount returns -1 and neither of the above two statements are true so we get the existing behaviour.

So the only time the behaviour of the plugin changes is if someone explicitly sets the count/badge to 0 in the push payload? Note, that seems to be the best way forward.

Reply:
That's my understanding, yes.

Note that if a title or message is also supplied then a new notification is created, as per current behaviour.

And the event handler is still called, as per current behaviour.

Reply:
Sweet that's what I was hoping for 🤞 I'll be able to test soon.